port/unix/omrmmap.c

```diff
@@ -50,11 +49,99 @@
 #include "protect_helpers.h"
 #include "omrportpriv.h"
 
+#if defined(J9ZOS390)
+#include "omrvmem.h"
+#if !defined(MAP_FAILED)
+#define MAP_FAILED ((void *)(intptr_t)-1)
+#endif /* !defined(MAP_FAILED) */
+#endif /* defined(J9ZOS390) */
+
 #if defined(AIXPPC)
 #include <sys/shm.h>
 #include <sys/vminfo.h>
 #endif/*AIXPPC*/
 
+#if defined(J9ZOS390)
+static J9MmapHandle *
+omrmmap_read_map_file(struct OMRPortLibrary *portLibrary, intptr_t file, uint64_t offset, uintptr_t size, uint32_t category)
+{
+	/* default implementation will allocate memory and read the file into it */
+	uintptr_t numBytesRead = 0;
+	intptr_t rc = 0;
+	void *mappedMemory = NULL;
+	void *allocPointer = NULL;
+	J9MmapHandle *returnVal = NULL;
+
+	Trc_PRT_mmap_map_file_default_entered(file, size);
+
+	if (-1 == file) {
+		Trc_PRT_mmap_map_file_default_badfile();
+		return NULL;
+	}
+
+	if (0 == size) {
+		struct stat buf;
+
+		memset(&buf, 0, sizeof(buf));
+		if (-1 == fstat(file - FD_BIAS, &buf)) {
+			Trc_PRT_mmap_map_file_unix_filestatfailed();
+			portLibrary->error_set_last_error(portLibrary, errno, OMRPORT_ERROR_MMAP_MAP_FILE_STATFAILED);
+			return NULL;
+		}
+		size = buf.st_size;
+	}
+
+	if (0 != portLibrary->file_seek(portLibrary, file, offset, EsSeekSet)) {
+		Trc_PRT_mmap_map_seek_failed(offset);
+		return NULL;
+	}
+
+	/**
+	 * The mappedMemory will point to an address that is 8-byte aligned.
+	 * Add 8 to size so that the allocated bytes are always greater than the requested size after mappedMemory.
+	 */
```

Comment:
I think we should remove the extra code here that tries to produce an aligned address: I think `portLibrary->mem_allocate_memory()` must already do that.

Reply:
I agree with removing it. When Keith and I discussed it, Keith referred to https://en.cppreference.com/w/c/types/max_align_t

Reply:
Removed.